src/DependencyInjection/Configuration.php

```diff
@@ -19,10 +19,12 @@ class Configuration implements ConfigurationInterface
     public const TRACEMODE_TRACECONTEXT = 'tracecontext';
     public const TRACEMODE_TRACEID      = 'traceid';
 
+    /**
+     * @return TreeBuilder<'array'>
+     */
     public function getConfigTreeBuilder(): TreeBuilder
```

Comment:
this works for sf7 & sf8 now without any issues, but the tricky part is with sf6 - here generics are not defined on `TreeBuilder` & `ArrayNodeDefinition` :thinking: so thats a bit tricky now.